pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -144,7 +148,16 @@ func (srv *ProvisioningSrv) RouteGetContactPointsExport(c *contextmodel.ReqConte
 		OrgID:   c.SignedInUser.GetOrgID(),
 		Decrypt: c.QueryBoolWithDefault("decrypt", false),
 	}
-	cps, err := srv.contactPointService.GetContactPoints(c.Req.Context(), q, c.SignedInUser)
+
+	if q.Decrypt {
+		action := accesscontrol.ActionAlertingProvisioningReadSecrets
+		evaluator := accesscontrol.EvalPermission(action)
+		if !accesscontrol.HasAccess(srv.ac, c)(evaluator) {
+			return response.Err(authz.NewAuthorizationErrorWithPermissions("read contact point secrets", evaluator))
+		}
+	}
```

Comment:
What do you think about adding an abstraction level for authz, and introducing a ContactPointAuthzService (interface in provisioning package with methods CanReadSecrets(user)) and a struct that will encapsulate accesscontrol.Accesscontrol and implement the interface in ngalert's accesscontrol package?

Reply:
It's upcoming, either along with or after adding scoped receiver permissions. I don't know that it necessarily makes sense to start that here, given it would be a decent amount of extra code to do the same thing and we can just as easily extract this later to that new authz service.